### PR TITLE
Make appeal view, claims search, and tasks polymorphic

### PR DESCRIPTION
**Pull request by mdbenjam**
Connects #5753

### Description
Now that we have appeals and legacy appeals we need to make DB tables that will have relations to both, polymorphic.

### Testing Plan
1. Have migrated up with data.
2. Have migrated down with data.

### CODE REVIEW

File: app/controllers/reader/claims_folder_searches_controller.rb [hunk 1]
```diff
@@ -12,6 +12,6 @@ def create
   private
 
   def appeal
-    LegacyAppeal.find_or_create_by_vacols_id(params[:appeal_id])
+    Appeal.find_appeal_by_id_or_find_or_create_legacy_appeal_by_vacols_id(params[:appeal_id])
```
Comment (shanear) [inline app/controllers/reader/claims_folder_searches_controller.rb]: lol at this method name. There's got to be a better name

File: app/models/legacy_appeal.rb [hunk 2]
```diff
@@ -7,6 +7,9 @@ class LegacyAppeal < ApplicationRecord
   belongs_to :appeal_series
   has_many :dispatch_tasks, foreign_key: :appeal_id, class_name: "Dispatch::Task"
   has_many :worksheet_issues, foreign_key: :appeal_id
+  has_many :appeal_views, as: :appeal
+  has_many :claims_folder_search, as: :appeal
+  has_many :task, as: :appeal
```
Comment (shanear) [inline app/models/legacy_appeal.rb]: Are these associations required for the polymorphic association to work properly? I don't think so, but im not sure?

File: db/migrate/20180626151855_remove_appeal_view_index.rb [hunk 3]
```diff
@@ -0,0 +1,12 @@
+class RemoveAppealViewIndex < ActiveRecord::Migration[5.1]
+  disable_ddl_transaction!
+  safety_assured
```
Comment (shanear) [inline db/migrate/20180626151855_remove_appeal_view_index.rb]: Is safety really assured here? I imagine we will deploying this under relatively high traffic, and messing around with deleting/adding indexes might cause issues.

We may just want both indexes first and then delete the irrelevant index later.

File: db/migrate/20180626154654_backfill_appeal_default_type_value.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
+class BackfillAppealDefaultTypeValue < ActiveRecord::Migration[5.1]
```
Comment (shanear) [inline db/migrate/20180626154654_backfill_appeal_default_type_value.rb]: Are these migration timestamps such that they'll run in the correct order? It seems like they are but it's hard to tell and just want to make sure.

File: app/models/appeal.rb [hunk 5]
```diff
@@ -1,4 +1,8 @@
 class Appeal < AmaReview
+  has_many :appeal_views, as: :appeal
+  has_many :claims_folder_search, as: :appeal
+  has_many :task, as: :appeal
```
Comment (shanear) [inline app/models/appeal.rb]: again, unless we are using these methods, we don't really need to add these. I don't think they're required for the polymorphic association to work properly.

File: app/controllers/reader/claims_folder_searches_controller.rb (hunk 1 above)
Comment (mdbenjam) [reply app/controllers/reader/claims_folder_searches_controller.rb]: I know it's super long, but I thought we should be very explicit about what is happening.

File: db/migrate/20180626151855_remove_appeal_view_index.rb (hunk 3 above)
Comment (mdbenjam) [reply db/migrate/20180626151855_remove_appeal_view_index.rb]: Good idea, we can remove this in a follow up PR.

File: db/migrate/20180626154654_backfill_appeal_default_type_value.rb (hunk 4 above)
Comment (mdbenjam) [reply db/migrate/20180626154654_backfill_appeal_default_type_value.rb]: Yup we set the data first, then we say you can't have null columns.

**Comment by mdbenjam** [comment]
I'll split out the index changes into two more PRs.

File: app/models/appeal.rb (hunk 5 above)
Comment (mdbenjam) [reply app/models/appeal.rb]: Doesn't it just make it easier if we define them? Allows us to do nice rails magic stuff without having to think about it.

**Pull request merged**